pkg/recordset/recordset.go

```diff
@@ -320,5 +381,9 @@ func targetStackName(clusterName string) string {
 
 func extractClusterName(sourceStackName string) string {
 	parts := strings.Split(sourceStackName, "-")
-	return parts[1]
+	if len(parts) >= 2 {
+		return parts[1]
+	}
+
+	return ""
```

Comment:
I think it should return error. How is empty `sourceClusterName` handled?

Reply:
Good point!

Empty `sourceClusterName` is not handled and we should stop processing if we are not able to get it.
We already filter/validate name while fetching but good to be extra cautious there.

Reply:
fixed